api/handlers.go

```diff
@@ -52,6 +70,29 @@ func CreateSessionHandlerFunc(sessions Sessions) http.HandlerFunc {
 
 		w.Header().Set("Content-Type", "application/json")
 		w.WriteHeader(http.StatusCreated)
-		w.Write(sessJson)
+		w.Write(sessJSON)
+	}
+}
+
+func GetSessionHandlerFunc(cache Cache, getVarsFunc GetVarsFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		ID := getVarsFunc(r)["ID"]
+
+		log.Event(ctx, fmt.Sprintf("get session by ID: %s", ID), log.INFO)
+
+		result, err := cache.Get(ID)
```

Comment:
You need to add unit tests to cover cases where:
- No session is found.
- Get session returns an error.

Reply:
got one for no session found, not sure about how to cover one for when a get session returns an error.